### PR TITLE
implement vectorized evaluation for builtinRightBinarySig 

### PR DESCRIPTION
**Pull request by icditwang**
### What problem does this PR solve? 
implement vectorized evaluation for builtinRightBinarySig 

### What is changed and how it works?
`
BenchmarkVectorizedBuiltinStringFunc/builtinRightBinarySig-VecBuiltinFunc-8         	  100000	     15726 ns/op	       0 B/op	       0 allocs/op

BenchmarkVectorizedBuiltinStringFunc/builtinRightBinarySig-NonVecBuiltinFunc-8      	   30000	     40079 ns/op	       0 B/op	       0 allocs/op`

### Check List 

Tests 

 - Unit test

### CODE REVIEW

File: expression/builtin_string_vec.go [hunk 1]
```diff
@@ -1376,11 +1376,44 @@ func (b *builtinFormatSig) vecEvalString(input *chunk.Chunk, result *chunk.Colum
...
+		return err
+	}
+	defer b.bufAllocator.put(buf2)
+	if err := b.args[1].VecEvalInt(b.ctx, input, buf2); err != nil {
+		return err
+	}
+	right := buf2.Int64s()
+	result.ReserveString(n)
+	for i := 0; i < n; i++ {
+		if buf.IsNull(i) {
```
Comment (wshwsh12) [inline expression/builtin_string_vec.go]: `buf.isnull(i) | buf2.isnull(i)`

File: expression/builtin_string_vec.go [hunk 2]
```diff
@@ -1376,11 +1376,44 @@ func (b *builtinFormatSig) vecEvalString(input *chunk.Chunk, result *chunk.Colum
...
+			result.AppendNull()
+			continue
+		}
+		str, rightLength := buf.GetString(i), int(right[i])
```
Comment (wshwsh12) [inline expression/builtin_string_vec.go]: Why transform right[i] (`int64`) to `int`?

Comment (icditwang) [reply expression/builtin_string_vec.go]: The non-vec func used it.It maybe wrong if we don't use it when we run the func on 32-bit length machines?

**Pull request merged**